zerver/views/users.py

```diff
@@ -304,6 +315,7 @@ def add_bot_backend(request, user_profile, full_name_raw=REQ("full_name"), short
                                      base_url=payload_url,
                                      interface=interface_type,
                                      token=random_api_key())
+    notify_created_bot(bot_profile)
```

Comment:
@timabbott Yes! That is exactly the issue here and the necessity why we need this. 

As the server event for the bot thus created is already pushed in `do_create_user`, any function/method using service object related to this bot will be in effect only after a refresh.